internal/datastore/revisions/hlcrevision.go

```diff
@@ -3,19 +3,20 @@ package revisions
 import (
 	"time"
 
+	"github.com/cockroachdb/apd"
 	"github.com/shopspring/decimal"
 
 	"github.com/authzed/spicedb/pkg/datastore"
 )
 
 // HLCRevision is a revision that is a hybrid logical clock, stored as a decimal.
 type HLCRevision struct {
-	decimal decimal.Decimal
+	decimal *apd.Decimal
```

Comment:
why a pointer here? if we directly embed, then an array of hlcs gc just has to check 1 object isntead of all of the entires.

Reply:
All the New* methods return a pointer and all the accessors on it are on the pointer type

Reply:
Changed to grab the value and use that